Ooniprobe 3.x.x

hellais opened the pull request.

hellais committed:
Upgrade ooniprobe formula to 3.0.9

hellais committed:
Add config file path to the manual run option

hellais committed:
Replace etc with HOMEBREWPREFIX_ETC

hellais committed:
Bump to ooniprobe 3.0.10

bassosimone commented on Formula/ooniprobe.rb:
```diff
@@ -3,287 +3,146 @@ class Ooniprobe < Formula
 
-    virtualenv_install_with_resources
+    cd "src/github.com/ooni/probe-cli" do
+      system "go", "build", "-o", bin/"ooniprobe", "./cmd/ooniprobe"
```
I would recommend to use `./build.sh macos` so that we use the linked flags for shrinking the binary.

hellais replied:
fixed!

bassosimone commented on Formula/ooniprobe.rb:
```diff
@@ -3,287 +3,146 @@ class Ooniprobe < Formula
+        "send_crash_reports": true,
+        "collect_usage_stats": true,
+        "collector_url": "",
+        "bouncer_url": "https://bouncer.ooni.io"
```
I'd rather not add here any bouncer or collector URL. I think they're not supported anymore and in any case we want `ooniprobe` to automatically choose. For the same reason I would remove the domain fronting key.

hellais replied:
fixed!

bassosimone commented on Formula/ooniprobe.rb:
```diff
@@ -3,287 +3,146 @@ class Ooniprobe < Formula
+    (pkgshare/"etc/ooniprobe-daily-config.json").write <<~EOS
+    {
+      "_version": 3,
+      "_informed_consent": true,
```
How are we going to obtain informed consent from the user?

hellais replied:
fixed!

hellais replied:
This will bypass it only in the case of the systemctl service which the user has to manually run a command to enable. I added a caveat in the homebrew package which will display the informed consent text right above the system enable command.

I assume power users are running this which are agreeing to it by virtue of running the system command.

hellais replied:
![Screenshot 2020-11-26 at 12 35 29](https://user-images.githubusercontent.com/424620/100346308-e91cea00-2fe3-11eb-8862-f169cf50ba5f.png)
This is how the informed consent looks like when installing in homebrew.

bassosimone commented on Formula/ooniprobe.rb:
```diff
@@ -3,287 +3,146 @@ class Ooniprobe < Formula
-          <true/>
+
+        <key>StartInterval</key>
+        <integer>3600</integer>
```
Can we make this interval like six hours, at least for the first release?

hellais replied:
fixed!

bassosimone commented on Formula/ooniprobe.rb:
```diff
@@ -3,287 +3,146 @@ class Ooniprobe < Formula
-      <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
-      <plist version="1.0">
-      <dict>
+    <?xml version="1.0" encoding="UTF-8"?>
```
Is this the user's plist? I'm wondering what happens if the user manually runs `ooniprobe` while an automatic test is running? I think there will be conflict with locking the database? Maybe we want to use a specifically different OONI HOME for this kind of use cases?

hellais replied:
fixed!

hellais replied:
They use different OONI_HOME directories now.

bassosimone reviewed (changes requested):
Thank you @hellais for working on this! I've some request for changes and a bunch of hopefully useful comments for improving this already-very-good PR! 🎸

hellais committed:
Run full list daily

hellais committed:
Address comments from @bassosimone

hellais committed:
Cleaner formatting

hellais committed:
Use var/lib

hellais committed:
Align informed consent with https://github.com/ooni/probe-cli/pull/162/files

hellais committed:
Fix typo in build command

hellais committed:
Drop unsused settings

hellais committed:
Create HOMEBREW_PREFIX/var/lib/ooniprobe

hellais committed:
Write everything to {var}

hellais committed:
Edit default command

hellais committed:
various path related fixes

hellais committed:
Log to /tmp by default for debugging

hellais committed:
Write config to var

hellais committed:
Bump to ooniprobe 3.0.11

hellais committed:
Install config to prefix/share

hellais committed:
Bump revision

hellais committed:
Fix typo in share path

bassosimone committed:
fix ooniprobe formula and be ~ready for opening PR

1. only run the unattended test group (so we don't run performance basically)

2. use the syslog log handler so logs go into the system log

3. fix the way in which we install the service so it runs in the context
of the user's login session, which is easier to handle

4. use the state directory as the working directory

This currently only works with master so we need to cut a release before
we can open a PR with upstream.

Reference issue: https://github.com/ooni/probe/issues/1289

hellais committed:
Bump probe-cli version

bassosimone commented:
Superseded by https://github.com/Homebrew/homebrew-core/pull/69399.
